### ISSUE TITLE
What's the different between neural-compressor and NNCF?

### ISSUE THREAD

**Issue by sanbuphy** [intel/neural-compressor#643]
I want to know the difference between them, thanks.

**Comment by NeoZhangJianyu** [comment]
@sanbuphy 
Intel Neural Compressor is a separated product to focus on optimizing AI models (TF/Pytorch/MXNet/ONNX) inference and it's neutral for AI frameworks (TF/Pytorch/MXNet/ONNX).

It is tracing the new technologies to speed up the AI model in inference.
For example, it supports more [quantization methods](https://github.com/intel/neural-compressor/blob/master/docs/source/quantization.md).

It also cooperates with Intel Extension for Pytorch/Tensorflow for Intel CPU/GPU.

Which AI framework are you using?

**Comment by sanbuphy** [comment]
> @sanbuphy Intel Neural Compressor is a separated product to focus on optimizing AI models (TF/Pytorch/MXNet/ONNX) inference and it's neutral for AI frameworks (TF/Pytorch/MXNet/ONNX).
> 
> It is tracing the new technologies to speed up the AI model in inference. For example, it supports more [quantization methods](https://github.com/intel/neural-compressor/blob/master/docs/source/quantization.md).
> 
> It also cooperates with Intel Extension for Pytorch/Tensorflow for Intel CPU/GPU.
> 
> Which AI framework are you using?

Hi @NeoZhangJianyu 
I'm using pytorch/paddle/onnx.
I have been using this tool （https://github.com/openvinotoolkit/nncf）which has POT quantization, but I'm not sure how it differs from Intel® Neural Compressor. Which one is better and easier to use?

**Comment by NeoZhangJianyu** [comment]
@sanbuphy 
NNCF is a tool of OpenVINO. 
As you are using Pytorch/ONNX, I suggest you try Neural Compressor. :)


In general, we want to quantize a model to get better performance and less accuracy loss.
If a tool can't match your requirement, you could try another.

For usage, both them are easy to use.

For performance, it's depended on the integrated algorithms and the feature of your model.
I think neural compressor is good in most cases.

Neural compressor has special developers for Pytorch, ONNX model. And there are many verified models of Pytorch & ONNX in daily test work.

Neural compressor also cooperate with Intel Pytorch team and ONNX support team for special cases of customers on Intel CPU & GPU.

So, I suggest to try Neural Compressor. :)

**Comment by sanbuphy** [comment]
> @sanbuphy NNCF is a tool of OpenVINO. As you are using Pytorch/ONNX, I suggest you try Neural Compressor. :)
> 
> In general, we want to quantize a model to get better performance and less accuracy loss. If a tool can't match your requirement, you could try another.
> 
> For usage, both them are easy to use.
> 
> For performance, it's depended on the integrated algorithms and the feature of your model. I think neural compressor is good in most cases.
> 
> Neural compressor has special developers for Pytorch, ONNX model. And there are many verified models of Pytorch & ONNX in daily test work.
> 
> Neural compressor also cooperate with Intel Pytorch team and ONNX support team for special cases of customers on Intel CPU & GPU.
> 
> So, I suggest to try Neural Compressor. :)


Thank you for your recommendation!

Actually, I found that precision explosion is a common problem in the quantization process. I wonder how neural-compressor  handles this phenomenon. Also, regarding neural-compressor , can it be used with openvino? 

I would like to ask that  can the quantized and pruned models be loaded and used  if I use the openvino inference framework?  openvino only changes the implementation and scheduling acceleration of operators.

Thank you very much.

**Comment by NeoZhangJianyu** [comment]
Neural-compressor support accuracy driven algorithm:
quantize model -> test performance and accuracy -> change quantize method/parameter ->
quantize model -> test performance and accuracy -> change quantize method/parameter
..
The loop is continued until find a right quantize mothed/parameter to match the goal.

No, it can't be used with OpenVINO.
INC can make sure the quantized Pytorch model has good performance and accuracy in Pytorch. 
But the result of converting it to IR (OpenVINO model format) is unknown, because quantized operators (like quantized-Conv2d) of Pytorch and OpenVINO can't be equaled in math result.
That will lead to the performance and accuracy are not same as original quantized model.

I suggest:
1. Convert FP32 AI model format in different frameworks, like pytorch to OpenVINO.
2. Quantize FP32 model to INT8 model in same AI framework, like Pytorch FP32 to Pytorch INT8.

**Comment by NeoZhangJianyu** [comment]
By the way, what's the hardware are you using?

You could find the examples of verified model list by INC in [examples](https://github.com/intel/neural-compressor/tree/master/examples).
If your model is in this list, you can refer the example code easily.

**Comment by sanbuphy** [comment]
> By the way, what's the hardware are you using?
> 
> You could find the examples of verified model list by INC in [examples](https://github.com/intel/neural-compressor/tree/master/examples). If your model is in this list, you can refer the example code easily.

I am using the CPU without integrated graphics, but I would like to leverage the power of Intel integrated graphics about quantization. Thank you for the recommendation, I will have a try.

**Comment by NeoZhangJianyu** [comment]
OK! Got it!
Good luck! :)
